src/cluster/Backend.h

```diff
@@ -410,6 +416,11 @@ class Backend {
      */
     void SetNodeId(std::string nid);
 
+    /**
+     * Provides access to the detail::Telemetry handle.
+     */
+    detail::Telemetry& Telemetry() const { return *telemetry; }
```

Comment:
Nit: `const` method should return `const`. Here we want a mutable value, so the getter should not be `const`. Let's also assert that `telemetry` is set.

```suggestion
    detail::Telemetry& Telemetry() {
        assert(telemetry);
        return *telemetry;
    }
```